lib/SILOptimizer/PassManager/PassPipeline.cpp

```diff
@@ -254,6 +254,11 @@ static void addMandatoryDiagnosticOptPipeline(SILPassPipelinePlan &P) {
   P.addMandatoryPerformanceOptimizations();
   P.addOnoneSimplification();
   P.addInitializeStaticGlobals();
+
+  if (P.getOptions().EmbeddedSwift) {
```

Comment:
Can you add a comment here why we need this in embedded swift, but don't want to do it in regular swift (like you said in your reply below).